src/AbstractAlgebra.jl

```diff
@@ -245,6 +245,13 @@ macro show_name(io, O)
 end
 
 function find_name(A, M = Main)
+  if M === Main && isdefined(Main, :CurrentModule)
+    a = find_name(A, Main.CurrentModule)
+    if a !== nothing
+      @show :bingo
```

Comment:
This PR is in principle what I had in mind, though of course this `@show` should go, and an actual example showcasing how to use this be inserted.